### DISCUSSION TITLE
MV2 / MV3 discussion

### DISCUSSION THREAD

**Discussion by ryanbr** [uBlockOrigin/uAssets#14544] [General]
With upcoming MV3 changes incoming from Google, can we get a MV2 uBO/uMatrix builds for any chromium-based browsers not transitioning to MV3 post January?

**Comment by MasterKia** [comment]
> any chromium-based browsers not transitioning to MV3

Which browsers have stated that they won't transition to MV3?
IIRC Brave and Vivaldi.

Google Chrome and Microsoft Edge are transitioning to MV3 and I assume Opera will do the same.

Vivaldi has its own built-in blocker, so should/would they care?

Brave has its own built-in blocker written in Rust and uses uBO's filter, so would it really be beneficial to keep supporting MV2?

**Comment by BrendanEich** [comment]
uBO does more by default, and has more knobs for its users, than Brave shields has or will likely ever have.

I've also committed to supporting uMatrix on Brave. We don't plan to build its functionality in. But we do support good extensions that require V2.

Browsers should not break such extensions with the only recourse being some browsers trying to absorb all the good (for their users, opinions vary) things those extensions do into their built-in features and options. I'm sure we agree on the first "should": Google shouldn't break V2, but I wanted to put out the second as a normative statement.

We (I was there) pioneered add-ons in Firefox for several reasons: to declutter built-in UX and underlying code; to encourage green-field extension development, to go beyond what we could do as a small team; to foster a new tier of innovation between browser and website. Harvesting best ideas back into baseline always an option but never a requirement.

I'm of same mind now running Brave as I was running tech side of Firefox: we want good extensions to endure and prosper. HTH

**Reply by MasterKia** [reply]
In the long run, would it be technically challenging for the Brave team to keep maintaining the old MV2 code in Chromium?

I don't know if Google **_is able_** to introduce new changes that would massively prevent you from keeping the MV2 support.

**Reply by krystian3w** [reply]
Enough will be block MV2 in shop (on 31 december 2022) and manually install on June 2023 with Corporation polices.

**Reply by BrendanEich** [reply]
> In the long run, would it be technically challenging for the Brave team to keep maintaining the old MV2 code in Chromium?
> 
> I don't know if Google **_is able_** to introduce new changes that would massively prevent you from keeping the MV2 support.

Please see https://twitter.com/BrendanEich/status/1534905779630661633. We'll keep supporting V2 until it gets too costly for us. If they remove it completely soon after their current deadline, I'd be surprised — but who knows with Google?

**Reply by mapx-** [reply]
@BrendanEich 
If brave could implement **_html filtering_** in its RUST-adblocking engine (html filtering only works in firefox due to chrome limitations) it would be a very important step (limiting the need to maintain the mv2 API in brave)
https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#html-filters

**Reply by BrendanEich** [reply]
@mapx- Even if we implemented html-filters, we'd still need MV2 for other stuff, right? Thanks for further clarification. My focus here is not to lose uBO on Brave after Chrome rug-pulls but while we still support V2.

**Comment by MasterKia** [comment]
> The first ad blocker built on the new Chrome's Manifest V3

https://www.producthunt.com/posts/adguard-mv3

https://adguard.com/en/blog/adguard-mv3.html

**Reply by krystian3w** [reply]
How checked is first place?

**Comment by ryanbr** [comment]
Any movement on this? 

Knowing other chromium forks Opera, Vivaldi (and Brave) are also wanting to keep MV2 supported. Can we have a MV2 Chromium build for those who still support it?

**Comment by gorhill** [comment]
There won't be MV3 build for uBO, it will continue as is. For MV3, it's uBOL.

**Reply by krystian3w** [reply]
I mean he asking to don't delete/disable Chromium folder in repository.

**Reply by uBlock-user** [reply]
Will uBOL receive dynamic filtering feature ?

**Comment by Yuki2718** [comment]
@gorhill 
> Can we have a MV**2** Chromium build for those who still support it?

(emphasis mine)

**Comment by gorhill** [comment]
Yes, Chromium build of uBO will still be released.

**Reply by ryanbr** [reply]
For those using MV2 still in Chromium forks, that is good

**Comment by ghost** [comment]
BTW kiwibrowser is supporting mv2 and announced it wont discontinue mv2 even after chrome droppped it, because it wont be much extra trouble as the whole support for extensions is done by dev of kiwibrowser. Strangly enough the ublock lite extension is not supported in kiwibrowser so I really hope mv2 chrome continues

**Reply by Yuki2718** [reply]
> kiwibrowser is supporting mv2 and announced it wont discontinue mv2 even after chrome droppped it

Source?

**Reply by krystian3w** [reply]
Most likely it has from the very beginning collected files to run WebExtension.

So it only has to test the quality of the code and whether it will hit some ugly code in the mobile version of the browser around 2024 (corporation sabotage on small projects).

**Reply by ghost** [reply]
> > kiwibrowser is supporting mv2 and announced it wont discontinue mv2 even after chrome droppped it
> 
> Source?
https://twitter.com/kiwibrowser/status/1591019650087030784
https://www.reddit.com/comments/q2cs0a/comment/hfmvv8u?context=3 response of dev

**Reply by Yuki2718** [reply]
Is it possible to sideload extension to Kiwi? As uBO will disappear from Chrome store on EOL of MV2.

**Comment by mikhoul** [comment]
[**@
            Yuki2718
**](https://github.com/Yuki2718) commented on [Feb 1, 2023, 6:56 AM EST](https://github.com/uBlockOrigin/uAssets/discussions/14544#discussioncomment-4839844 "2023-02-01T11:56:52Z - Replied by Github Reply Comments"):
> Is it possible to sideload extension to Kiwi? As uBO will disappear from Chrome store on EOL of MV2.

Yes it's possible.

**Reply by krystian3w** [reply]
Who hide poll: ![obraz](https://user-images.githubusercontent.com/35370833/216165437-7ddad5a8-6f22-485c-8686-063829ae51b0.png)

**Comment by ryanbr** [comment]
Is it possible for the next release, possible to create a static MV2 crx hosted on Github (with "update url" pointing to https://github.com/gorhill/uBlock/releases ). ?

cc: @gorhill

**Reply by MasterKia** [reply]
https://github.com/uBlockOrigin/uBlock-issues/discussions/3514#discussioncomment-11866988

**Reply by u-RraaLL** [reply]
@ryanbr Have you guys come up with a solution to keeping uBO on Brave when the crx method doesn't work?

Stable Chromium 139 without MV2 is set to release on Aug 5. Extension removal from CWS will probably follow shortly after\* so there's about 2 more months to come up with a solution.

I'm asking because I'd like to update the reddit megathread about the issue and have most recent info from all sides covered.

---

Edit:

\* Theoretically, CWS could remove MV2 extensions even before Chromium 139 stable release. So maybe there's less time after all.

**Reply by Yuki2718** [reply]
You have to enable via Brave's settings (Extensions > Manifest V2 extensions).

**Reply by u-RraaLL** [reply]
I know that, but this installs from CWS. Ryan asked for the crx above to change that for future releases (I'm assuming).

---

If CWS removed extensions today, only Opera would be left (I'm not counting the outdated Edge version, that will likely also disappear soon).
